### PR TITLE
log additional memory information

### PR DESCRIPTION
**Pull request by owaddell**
**Proposed changes**

* log total physical memory in `<system-attributes>` via `uv_get_total_memory`
* log free memory in `<statistics>` via `uv_get_free_memory`
* add free and total physical memory to the diagnostics debug page

### CODE REVIEW

File: doc/swish/osi.tex [hunk 1]
```diff
@@ -310,6 +310,25 @@
 function. On macOS, it calls the \code{mstats} function. On Windows,
 it calls the \code{\_heapwalk} function.
 
+\defineentry{osi\_get\_free\_memory}
+\begin{procedure}
+  \code{(osi\_get\_free\_memory)}
+\end{procedure}
+\returns{} an unsigned integer
+
+Returns the number of bytes of free memory as reported by
```
Comment (burgerrg) [inline doc/swish/osi.tex]: For consistency, "The \code{osi\\_get\\_free\\_memory} function returns ..."

File: doc/swish/osi.tex [hunk 2]
```diff
@@ -310,6 +310,25 @@
...
+\code{uv\_get\_free\_memory}.
+
+\defineentry{osi\_get\_total\_memory}
+\begin{procedure}
+  \code{(osi\_get\_total\_memory)}
+\end{procedure}
+\returns{} an unsigned integer
+
+Returns the number of bytes of total physical memory as reported by
```
Comment (burgerrg) [inline doc/swish/osi.tex]: See comment on osi_get_free_memory.

File: doc/swish/statistics.tex [hunk 3]
```diff
@@ -108,6 +108,7 @@
   \argrow{gc-real}{elapsed time in seconds of garbage collections
     since last event}
   \argrow{gc-bytes}{Scheme heap bytes reclaimed since last event}
+  \argrow{os-free-memory}{current free memory from \code{osi\_get\_free\_memory} }
```
Comment (burgerrg) [inline doc/swish/statistics.tex]: Suggestion: remove extra space before final }

File: doc/swish/osi.tex (hunk 1 above)
Comment (owaddell) [reply doc/swish/osi.tex]: Fixed.

File: doc/swish/osi.tex (hunk 2 above)
Comment (owaddell) [reply doc/swish/osi.tex]: Fixed. Thanks!

File: doc/swish/statistics.tex (hunk 3 above)
Comment (owaddell) [reply doc/swish/statistics.tex]: Fixed.

**Comment by laqrix** [comment]
Question: Would it be wise to wrap the uv_* functions with osi_* wrappers so that if the type signatures of the uv_ functions ever change, the C compiler might notice and throw an error or warning?

**Comment by burgerrg** [comment]
Since the return type is uint64_t, I think it's unlikely for the libuv memory functions to change the return value in the foreseeable future. 2^64 is a gigantic amount of memory.

**Review by burgerrg — APPROVED** [review]

**Comment by owaddell** [comment]
I agree the return and argument types for these libuv functions are unlikely to change. I pushed a test that may alert us if they change the units.

I considered adding the new memory information to the default mat meta-data but decided against that since folks can do this with more finesse using `mat:add-annotation!`.

**Comment by owaddell** [comment]
Thanks for the reviews, gang!

**Pull request merged**